jina/types/document/__init__.py

```diff
@@ -516,20 +506,42 @@ def _update_ndarray(self, k, v):
         elif isinstance(v, NdArray):
             NdArray(getattr(self._pb_body, k)).is_sparse = v.is_sparse
             NdArray(getattr(self._pb_body, k)).value = v.value
-        elif scipy_installed and scipy.sparse.issparse(v):
-            from ..ndarray.sparse.scipy import SparseNdArray
 
-            self._update_sparse_ndarray(k=k, v=v, sparse_cls=SparseNdArray)
-        elif tensorflow_installed and isinstance(v, tensorflow.SparseTensor):
-            from ..ndarray.sparse.tensorflow import SparseNdArray
+        else:
+            from ... import JINA_GLOBAL
```

Comment:
can we refactor this into a private function?

Reply:
Do you mean the whole logic from line 511 up to line 541 ?

Reply:
yes, if possible?

Reply:
Done